rest/server_context.go

```diff
@@ -46,6 +47,21 @@ type ServerContext struct {
 	statsContext *statsContext
 	HTTPClient   *http.Client
 	replicator   *base.Replicator
+	sync.Mutex            // Concurrent Start and Stop CPU profiling requests could race updating cpuPprofFile.
```

Comment:
We want this lock to only affect access to the `cpuPprofFile` variable, so I don't think embedding the mutex makes sense here.

Reply:
Makes sense; A lock cpuPprofFileMutex has been added to protect cpuPprofFile from concurrent Start and Stop CPU profiling requests. Changes are available in latest revision.